auth/jwt_controller.go

```diff
@@ -21,18 +21,39 @@ import (
 // the "auth.jwt.secret" config entry.
 // The token is set to expire in the amount of seconds defined by
 // the "auth.jwt.expiry" config entry.
-func GenerateToken(id interface{}) (string, error) {
+func GenerateToken(username interface{}) (string, error) {
 	expiry := time.Duration(config.GetInt("auth.jwt.expiry")) * time.Second
 	now := time.Now()
 	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
-		"userid": id,
+		"userid": username,
 		"nbf":    now.Unix(),             // Not Before
 		"exp":    now.Add(expiry).Unix(), // Expiry
 	})
 
 	return token.SignedString([]byte(config.GetString("auth.jwt.secret")))
 }
 
+// GenerateTokenWithClaims generates a new JWT with custom claims.
+// The token is created using the HMAC SHA256 method and signed using
+// the "auth.jwt.secret" config entry.
+// The token is set to expire in the amount of seconds defined by
+// the "auth.jwt.expiry" config entry.
+func GenerateTokenWithClaims(username interface{}, claims jwt.MapClaims) (string, error) {
```

Comment:
I just realized, developers may want to name it differently, but we are using it to fetch user from db.
```
result := database.GetConnection().Where(column.Name+" = ?", claims["userid"]).First(user)
```
(https://github.com/go-goyave/goyave/blob/34861e6d93255a6ca61b43567ad192c5b2c7859b/auth/jwt.go#L71)

I think we should either keep it or provide some way to specify fetching algorithm :X

Reply:
Maybe we should instead introduce `sub` claim which is the standard one (https://tools.ietf.org/html/rfc7519#section-4.1)

Reply:
Interesting, I didn't know about this RFC. At first I was thinking about adding a field to `JWTAuthenticator` to allow users to customize the name of the claim used.

Reply:
Using the "sub" claim looks better than "userid", but I don't know what kind of effect it would have on developers relying on the feature if we changed the claim name used. If they are using the token outside of their Goyave application, it could break things. They are not really supposed to do that with the current implementation, but it's a possibility. What do you think?

Reply:
I would keep both and only remove it when there is new major release. Also I would go back to the implementation I proposed initially (not allow user to override the mandatory claims) or mention that `sub` is required.

Reply:
Let's keep `userid` for now. And we will replace it with `sub` in the next major version. I'd like to keep the possibility for the developer to override claims if he wants to, that adds some flexibility. For example, developers may want to generate the `nbf` and `exp` claims dynamically.